pkg/controller/utils/machine_helper.go

```diff
@@ -0,0 +1,111 @@
+package utils
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	machineapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	awsproviderapi "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = logf.Log.WithName("machine_helper")
+
+func RemoveLBFromMasterMachines(kclient client.Client, elbName string, masterNodes *machineapi.MachineList) error {
+	for _, machine := range masterNodes.Items {
+		providerSpecDecoded, err := getDecodedProviderSpec(machine)
+		if err != nil {
+			log.Error(err, "Error retrieving decoded ProviderSpec for machine", "machine", machine.Name)
+			return err
+		}
+		lbList := providerSpecDecoded.LoadBalancers
+		newLBList := []awsproviderapi.LoadBalancerReference{}
+		for _, lb := range lbList {
+			if !strings.HasPrefix(elbName, lb.Name) {
+				log.Info("Machine's LB does not match LB to remove", "Machine LB", lb.Name, "LB to remove", elbName)
+				log.Info("Keeping machine's LB in machine object", "LB", lb.Name, "Machine", machine.Name)
+				newLBList = append(newLBList, lb)
+			}
+		}
+		err = updateLBList(kclient, lbList, newLBList, machine, providerSpecDecoded)
+		if err != nil {
+			log.Error(err, "Error updating LB list for machine", "machine", machine.Name)
+			return err
+		}
+	}
+	return nil
+}
+
+func AddLBToMasterMachines(kclient client.Client, elbName string, masterNodes *machineapi.MachineList) error {
+	for _, machine := range masterNodes.Items {
+		providerSpecDecoded, err := getDecodedProviderSpec(machine)
+		if err != nil {
+			log.Error(err, "Error retrieving decoded ProviderSpec for machine", "machine", machine.Name)
+			return err
+		}
+		lbList := providerSpecDecoded.LoadBalancers
+		newLBList := []awsproviderapi.LoadBalancerReference{}
+		for _, lb := range lbList {
+			if lb.Name != elbName {
+				newLBList = append(newLBList, lb)
+			}
+		}
+		newLB := awsproviderapi.LoadBalancerReference{
+			Name: elbName,
+			Type: awsproviderapi.NetworkLoadBalancerType,
+		}
+		newLBList = append(newLBList, newLB)
+		err = updateLBList(kclient, lbList, newLBList, machine, providerSpecDecoded)
+		if err != nil {
+			log.Error(err, "Error updating LB list for machine", "machine", machine.Name)
+			return err
+		}
+	}
+	return nil
+}
+
+func getDecodedProviderSpec(machine machineapi.Machine) (*awsproviderapi.AWSMachineProviderConfig, error) {
```

Comment:
Less important for non-exported functions/methods, but might still be useful.

Also, this is specific to AWS, so does it make sense to name the function to follow suit?